File: lab3/src/main/java/lab3/myquest/locations/Ceil.java
package lab3.myquest.locations;

import lab3.functionality.Location;
import lab3.functionality.Player;

public class Ceil extends Location {
    public Ceil(String name) {
        super(name, p -> p.getName() + " стал недолюбливать этот потолок.");
    }

    @Override
    public String enter(Player player) {
        player.damage();
        return player.getName() + " ударился головой об потолок.";
    }
}

File: lab4/src/main/kotlin/lab4/myQuest/locations/Ceil.kt
package lab4.myquest.locations

import lab4.functionality.Location
import lab4.functionality.Player

class Ceil() : Location("Потолок", { it.name + " стал недолюбливать этот потолок." }) {
    override fun enter(player: Player): String {
        player.damage()
        return player.name + " ударился головой об потолок."
    }
}